Sources/XCMetricsBackendLib/Statistics/Repositories/SQLStatisticsRepository.swift

```diff
@@ -0,0 +1,113 @@
+import NIO
+
+private struct CountResult: Decodable {
+    var builds: Int;
+    var errors: Int;
+}
+
+class SQLStatisticsRepository: StatisticsRepository {
+
+    let db: Database
+
+    init(db: Database) {
+        self.db = db;
+    }
+
+    func getBuildStatuses(page: Int, per: Int, using eventLoop: EventLoop) -> EventLoopFuture<Page<BuildStatusResult>> {
+        return Build.query(on: self.db)
+            .field(\.$id)
+            .field(\.$buildStatus)
+            .sort(\.$startTimestampMicroseconds, .descending)
+            .paginate(PageRequest(page: page, per: per))
+            .map { $0.map { BuildStatusResult(id: $0.id!, buildStatus: $0.buildStatus) } }
+    }
+
+    func getDayCounts(from: Date, to: Date, using eventLoop: EventLoop) -> EventLoopFuture<[DayCount]> {
+        return DayCount.query(on: db)
+            .filter(\.$id >= from)
+            .filter(\.$id <= to)
+            .sort(\.$id)
+            .all()
+            .flatMap { counts in
+                return eventLoop.makeSucceededFuture(
+                    self.fillDayCounts(counts: counts, from: from, to: to)
+                )
+            }
+
+    }
+
+    func getCount(day: Date, using eventLoop: EventLoop) -> EventLoopFuture<DayCount> {
+        guard let sql = db as? SQLDatabase else {
+            return eventLoop.makeFailedFuture(RepositoryError.unexpected(message: "The database is not SQL"))
+        }
+
+        let query: SQLQueryString = """
+        SELECT
+            sum(error_count) as errors,
+            count(*) as builds
+        FROM
+            \(raw: Build.schema)_\(raw: day.xcm_toPartitionedTableFormat())
+        """
+
+        return sql.raw(query)
+            .first(decoding: CountResult.self)
+            .flatMapAlways { result in
+                let count = (try? result.get()) ?? CountResult(builds: 0, errors: 0)
+                return eventLoop.makeSucceededFuture(DayCount(day: day, builds: count.builds, errors: count.errors))
+            }
+    }
+
+    func createDayCount(day: Date, using eventLoop: EventLoop) -> EventLoopFuture<Void> {
+        return self.getCount(day: day, using: eventLoop).flatMap { count in
+            count.create(on: self.db)
+        }
+    }
+
+    // MARK: - Private Methods
+
+    /// Since days without any build information potentially could exist (for instance if a job has failed or not been run)
+    /// we need to fill days without build information with zero values to keep the format consistent.
+    /// Performance of this method should be considered for large date ranges.
+    private func fillDayCounts(counts: [DayCount], from: Date, to: Date) -> [DayCount] {
+        let from = from.truncateTime()
+        let days = Calendar.current.dateComponents([.day], from: from, to: to.truncateTime()).day! + 1
+
+        guard days > 0 && counts.count != days else { return counts }
+
+        var filled = [DayCount]()
+
+        for dayOffset in stride(from: 0, to: days, by: 1) {
```

Comment:
Since you're step is 1, you can use a range here:
```suggestion
        for dayOffset in(0..<days) {
```